sdk/lib/cmlib/hiveinit.c

```diff
@@ -231,16 +355,23 @@ HvpInitializeMemoryHive(
     SIZE_T ChunkSize;
 
     ChunkSize = ChunkBase->Length;
-    DPRINT("ChunkSize: %zx\n", ChunkSize);
 
-    if (ChunkSize < sizeof(HBASE_BLOCK) ||
-        !HvpVerifyHiveHeader(ChunkBase))
+    if (ChunkSize < sizeof(HBASE_BLOCK))
     {
-        DPRINT1("Registry is corrupt: ChunkSize 0x%zx < sizeof(HBASE_BLOCK) 0x%zx, "
-                "or HvpVerifyHiveHeader() failed\n", ChunkSize, sizeof(HBASE_BLOCK));
+        DPRINT1("Registry is corrupt: ChunkSize 0x%zx < sizeof(HBASE_BLOCK) 0x%zx\n", ChunkSize, sizeof(HBASE_BLOCK));
         return STATUS_REGISTRY_CORRUPT;
     }
 
+    if (!HvpVerifyHiveHeader(ChunkBase, HFILE_TYPE_PRIMARY))
```

Comment:
(See also the comment below). On Windows, hives loaded with `HINIT_FLAT`, `HINIT_MEMORY` or `HINIT_MEMORY_INPLACE` don't perform the LOG restoration, they just do the header verification and bail out if it's invalid.
Code that loads hives with these flags manually run a fixup routine afterwards.

Only hives loaded with `HINIT_FILE` or `HINIT_MAPFILE` do the header restoration from log when loading with `HvInitialize()`